src/tools/clippy/tests/ui/missing_const_for_fn/cant_be_const.rs

```diff
@@ -71,14 +70,8 @@ mod with_test_fn {
     }
 }
 
-// Allowing on this function, because it would lint, which we don't want in this case.
-// if we have `#[start]` and `#[test]` check `is_entrypoint_fn(cx, def_id.to_def_id())` is stopped
-// working
-#[allow(clippy::missing_const_for_fn)]
-#[start]
-fn init(num: isize, something: *const *const u8) -> isize {
-    1
-}
```

Comment:
@rust-lang/clippy after resolving conflicts, this un-does some recent changes from https://github.com/rust-lang/rust-clippy/pull/13945. I am not sure what else to do here and don't quite follow the comment.

Reply:
Somewhat surprisingly this test passes even when the file does not have a main function...